parlai/scripts/convo_render.py

```diff
@@ -0,0 +1,201 @@
+            @page{{ margin: 0; size: {str(width)}in {str(height)}in; }}
+        }}
+        ul{{
+            list-style: none;
+            margin: 0;
+            padding: 0;
+        }}
+        ul li{{
+            display:inline-block;
+            clear: both;
+            padding: 20px;
+            border-radius: 30px;
+            margin-bottom: 2px;
+            font-family: Helvetica, Arial, sans-serif;
+        }}
+        .{other_speaker}{{
+                background: #eee;
+                float: left;
+            }}
+        .{human_speaker}{{
+            background: #0084ff;
+            color: #fff;
+            float: right;
+        }}
+        .breaker{{
+            color: #bec3c9;
+            display: block;
+            height: 20px;
+            margin: 20px 20px 20px 20px;
+            text-align: center;
+            text-transform: uppercase;
+        }}
+    </style>
+</head>
+<body>
+{gen_convo_ul(conversations)}
+</body>
+</html>
+    """
+    return html_str
+
+
+def pre_process(fname, num_ex, alt_speaker):
+    """
+    Pre-process the given file to bring the conversation in a certain format
+    :param fname: File name to be processed
+    :param num_ex: Number of conversations to be used
+    :param alt_speaker: Name of other speaker to be used
+
+    :return: List of tuples of the form: (speaker, speech)
+    """
+    conversation = []
+    with open(fname) as f:
+        lines = f.readlines()
+        random.shuffle(lines)
+        lines = lines[:num_ex]
+        for line in lines:
+            data = json.loads(line)
+            dialogue = data["dialog"]
+            for item in dialogue:
+                if item["speaker"] == "human_evaluator":
+                    speaker = "human"
+                else:
+                    speaker = alt_speaker
+                conversation += [(speaker, item["text"])]
+            conversation += [(END_OF_CONVO, END_OF_CONVO)]
+
+    return conversation
+
+
+def prBlueBG(text):
+    """
+    Print given in text with a blue background
+    :param text: The text to be printed
+    """
+    print("\033[44m{}\033[0m".format(text), sep="")
+
+
+def display_cli(conversations, alt_speaker, human_speaker):
+    """
+    Display the conversations on the Command Line
+    :param conversations: The dialogs to be displayed
+    :param alt_speaker: Name of other speaker to be used
+    :param human_speaker: Name of human speaker to be used
+    """
+    for speaker, speech in conversations:
+        if speaker == END_OF_CONVO:
+            print("-" * 20 + "END OF CONVERSATION" + "-" * 20)
+        elif speaker == alt_speaker:
+            print("%-15s: %s" % (speaker[:15], speech))
+        else:
+            prBlueBG("%-15s: %s" % (speaker[:15], speech))
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Process Conversation Rendering arguments"
+    )
+    parser.add_argument("i", help="Input file to read conversations from")
```

Comment:
generally better to use full names like "input" instead of "i". Especially since this will be used in `--help`.